client/components/plans/plans-compare/index.jsx

```diff
@@ -153,17 +154,17 @@ var PlansCompare = React.createClass( {
 	comparisonTable: function() {
 		var plansColumns,
 			featuresList = this.props.features.get(),
+			hideFreePlan = this.props.hideFreePlan ? true : false,
+			numberOfPlaceholders = 4,
 			plans = this.props.plans.get(),
-			site = this.props.selectedSite,
-			showJetpackPlans = site ? site.jetpack : false;
+			site = this.props.selectedSite;
 
 		plans = plans.filter( function( plan ) {
-			return ( showJetpackPlans === ( 'jetpack' === plan.product_type ) );
+			return filterPlansBySiteAndProps( plan, site, hideFreePlan );
```

Comment:
I did something similar in the end, but kept the function, so that we can share it between the components.